Tools/CISupport/ews-build/steps.py

```diff
@@ -565,6 +568,45 @@ def evaluateCommand(self, cmd):
         return rc
 
 
+class CheckOutPullRequest(steps.ShellSequence):
+    name = 'checkout-pull-request'
+    description = ['checking-out-pull-request']
+    descriptionDone = ['Checked out pull request']
+    haltOnFailure = True
```

Comment:
fine for now, but haltOnFailure would need to be false for Commit-Queue. 
Also you have both haltOnFailure=True as well as custom self.build.buildFinished in run().